### PR TITLE
fix(DropdownV2): match experimental spec

### PR DESCRIPTION
**Pull request by emyarod**
Closes IBM/carbon-components-react#2022

depends on
https://github.com/IBM/carbon-components/pull/2203
https://github.com/IBM/carbon-components/pull/2206/
https://github.com/IBM/carbon-components/pull/2212

This PR incorporates feedback from the March 2019 dropdown component audit

#### Changelog

**Changed**

- Integrate v10 listbox styles

**Removed**

- `type` prop for inline listbox components (requires refactoring of multiselect and combobox as well)

### CODE REVIEW

**Comment by emyarod** [comment]
https://github.com/IBM/carbon-components/pull/2203 only addressed the overflow issue for the trigger button label 🤦‍♂️ 

* need overflow styles on menu options

* inline list box menu options also need reduced side margins to match dropdown spec

image

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- [x] Just one change: The top rule line in the drawer is flush next to the focus state. It
Screen Shot 2019-03-26 at 3 25 10 PM
 needs to not be there when the focus border is active.

**Comment by emyarod** [comment]
tests are failing because of improperly formatted issue templates https://github.com/IBM/carbon-components-react/issues/2098

blocked until https://github.com/IBM/carbon-components-react/pull/2099 is merged in

**Review by laurenmrice — CHANGES_REQUESTED** [review]
**Changes:**

- [x] The disabled state shouldnt have a bottom rule .
Screen Shot 2019-03-28 at 10 13 13 AM

- [x] For the light variant there should call field-02 because it is the inverse.
Screen Shot 2019-03-28 at 10 26 55 AM

- [ ] I don't think the focus border should be triggered persistently upon selection of the label when you open the drawer. This is happening in Firefox, Safari, and Chrome and was not happening before.
Screen Shot 2019-03-28 at 10 28 33 AM

**Comment by emyarod** [comment]
regarding the last point, this seems to be the default behavior of Downshift, which we depend on for all of our listbox components. I opened https://github.com/IBM/carbon-components-react/issues/2111 to start the discussion about this

**Review by laurenmrice — APPROVED** [review]
Looks good to me ! 👍

File: src/components/ListBox/ListBoxMenuItem.js [hunk 1]
```diff
@@ -25,7 +26,16 @@ const ListBoxMenuItem = ({ children, isActive, isHighlighted, ...rest }) => {
   });
   return (
     <div className={className} {...rest}>
-      {children}
+      {componentsX ? (
+        <div
+          className={`${prefix}--list-box__menu-item__option`}
```
Comment (joshblack) [inline src/components/ListBox/ListBoxMenuItem.js]: Do we need keyboard handlers if this is `role="button"`? Or potentially another interactive prop like `onClick`?

Comment (emyarod) [reply src/components/ListBox/ListBoxMenuItem.js]: I believe that behavior is already handled by Downshift, these attributes are just holdovers from vanilla and can probably be removed altogether

File: src/components/DropdownV2/DropdownV2.js [hunk 2]
```diff
@@ -130,87 +147,115 @@ export default class DropdownV2 extends React.Component {
...
-                ))}
-              </ListBox.Menu>
-            )}
-          </ListBox>
-        )}
-      </Downshift>
-    );
-    return title || helper ? (
-      <>
+      <div className={wrapperClasses}>
```
Comment (asudoh) [inline src/components/DropdownV2/DropdownV2.js]: Given the change here - Wanted to see it's tested in `v9`. Thanks!

Comment (emyarod) [reply src/components/DropdownV2/DropdownV2.js]: I have tested it in v9 but I will rope it off as we don't have a wrapper in v9

**Review by joshblack — APPROVED** [review]

**Pull request merged**